src/components/elements/table/TableRowActions.tsx

```diff
@@ -1,51 +1,88 @@
-import { Stack } from '@mui/material';
-import { useMemo } from 'react';
+import { Button, Stack } from '@mui/material';
+import React, { ReactNode, useMemo } from 'react';
 import ButtonLink from '../ButtonLink';
 import CommonMenuButton, { CommonMenuItem } from '../CommonMenuButton';
 
 export type TableRowActionsType = {
-  primaryAction?: CommonMenuItem;
+  primaryAction?: CommonMenuItem | ReactNode;
   secondaryActions?: CommonMenuItem[];
 };
 
 interface TableRowActionsProps<T> {
   record: T;
   recordName?: string;
-  getActions: (record: T) => TableRowActionsType;
+  loading?: boolean;
+  getActions: (record: T, loading?: boolean) => TableRowActionsType;
 }
 
 const TableRowActions = <T extends { id: string }>({
   record,
   recordName,
   getActions,
+  loading,
 }: TableRowActionsProps<T>) => {
   const accessibleName = useMemo(
     () => recordName || record.id,
     [record.id, recordName]
   );
 
   const { primaryAction, secondaryActions } = useMemo(
-    () => getActions(record),
-    [getActions, record]
+    () => getActions(record, loading),
+    [getActions, loading, record]
+  );
+
+  const { primaryActionNode, primaryActionTypesafe } = useMemo(() => {
+    return React.isValidElement(primaryAction)
+      ? { primaryActionNode: primaryAction }
+      : { primaryActionTypesafe: primaryAction as CommonMenuItem };
+  }, [primaryAction]);
+
+  const secondariesWithAria = useMemo(
+    () =>
+      secondaryActions?.map((s) => {
+        return {
+          ...s,
+          ariaLabel: s.ariaLabel || `${s.title}, ${recordName}`,
```

Comment:
Do we need to support multiple ways of specifying the aria label? I wonder if we should just make it required on the primary/secondary action definitions, to simplify the logic.  (Instead of inferring them based on `recordName` and `title`)